### DISCUSSION TITLE
Glitches, jumps, interruptions in the recorded videos (RTSP stream)

### DISCUSSION THREAD

**Discussion by anonimusz20220712** [Motion-Project/motion#1634] [Q&A]
### Did you read the guide?

Yes

### What is the base version number of Motion being used?

4.3.x

### What was the install method?

Installed via package tool

### What is base architecture?

ARM-64bit

### What is the distro being used?

Raspbian

### Disto version number

Debian GNU/Linux 11 (bullseye)

### Camera type(s) being used?

Network camera with RTSP

### Describe the issue/problem and steps to reproduce

Dear Developers,

Recently I experience glitches and jumps often on the recorded footages. I have gigabit network at home, works well. I tested the system with openRTSP as well without problem. The source of motion's RTSP stream are two Tapo C110 cameras (there is a good quality, high speed Sandisk MicroSD inside in one of them); Both in 720p mode. In the past I didn't experience these problems. 

The problematic videos(previews):

https://drive.google.com/file/d/125TX9s8UnFj1V4Hix9vIxl5UCrLQ2Xk-/view?usp=sharing, https://drive.google.com/file/d/16iFLi3lnb5vp710BG4nd1SC4SDkluEYL/view?usp=sharing, https://drive.google.com/file/d/1Gp7pwHct63BKpX8F80jk2AqevO5Gp5G9/view?usp=sharing, https://drive.google.com/file/d/1HvN5JwdyzhyRmXeLOImD8jqtAe19vZdx/view?usp=sharing, https://drive.google.com/file/d/1JlBuln9uVJpFNegKKp80WHzPIi00dVIS/view?usp=sharing, https://drive.google.com/file/d/1LKMtz3OLZBLwJSv4r0jw5pc6s9gmLRKb/view?usp=sharing, https://drive.google.com/file/d/1R_U9FDY-JBfd5UxBzaIdOE0XTi3cbofl/view?usp=sharing, https://drive.google.com/file/d/1cjk5cIIYFzAK8qLzAdPPe-UIPf-VoblA/view?usp=sharing

Latest problematic raw file:
https://drive.google.com/file/d/10ItYM1H9Oua9aCXhaXYbQy5Q0HLVOMbE/view?usp=share_link

Thanks for your any help or advice!

Best Regards

Bela

![image](https://user-images.githubusercontent.com/124486842/216805580-2062ba7f-731a-42eb-84f2-b611734a5f01.png)

----
https://www.raspberrypi.com/products/raspberry-pi-4-model-b/specifications/
----
Raspberry Pi 4 Model B Rev 1.5 (4GB RAM)
----
cat /proc/version
Linux version 5.15.84-v8+ (dom@buildbot) (aarch64-linux-gnu-gcc-8 (Ubuntu/Linaro 8.4.0-3ubuntu1) 8.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #1613 SMP PREEMPT Thu Jan 5 12:03:08 GMT 2023
----
motion Version 4.3.2
Debian GNU/Linux 11 (bullseye)
----
Exif:
ExifTool Version Number         : 11.26
File Name                       : FOLYOSO_1-15-2023-02-02_11-08-33-Treshold-5000-ChangedPixels-59585-NoiseLevel-11.mkv
Directory                       : //rpi4smb-motion/motion
File Size                       : 8.8 MB
File Modification Date/Time     : 2023:02:02 11:09:32+01:00
File Access Date/Time           : 2023:02:05 07:01:18+01:00
File Creation Date/Time         : 2023:02:02 11:09:32+01:00
File Permissions                : rw-rw-rw-
File Type                       : MKV
File Type Extension             : mkv
MIME Type                       : video/x-matroska
EBML Version                    : 1
EBML Read Version               : 1
Doc Type                        : matroska
Doc Type Version                : 4
Doc Type Read Version           : 2
Timecode Scale                  : 1 ms
Muxing App                      : Lavf58.45.100
Writing App                     : Lavf58.45.100
Duration                        : 0:00:42
Track Number                    : 1
Track Language                  : und
Codec ID                        : V_MPEG4/ISO/AVC
Track Type                      : Video
Image Width                     : 1280
Image Height                    : 720
Display Unit                    : Unknown (4)
Tag Name                        : DURATION
Tag String                      : 00:00:42.372000000
Image Size                      : 1280x720
Megapixels                      : 0.922

### Relevant Motion log output (at log_level 8)

```shell
1:ml1:Folyoso] [NTC] [EVT] [Feb 02 09:59:18] event_newfile: File of type 8 saved to: /mnt/camera-storage/FOLYOSO_1-12-2023-02-02_09-59-17-Treshold-5000-ChangedPixels-14020-NoiseLevel-11.mkv
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 09:59:18] motion_detected: Motion detected - starting event 12
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 10:00:18] mlp_actions: End of event 12
[1:ml1:Folyoso] [NTC] [EVT] [Feb 02 10:14:12] event_newfile: File of type 8 saved to: /mnt/camera-storage/FOLYOSO_1-13-2023-02-02_10-14-11-Treshold-5000-ChangedPixels-7429-NoiseLevel-12.mkv
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 10:14:12] motion_detected: Motion detected - starting event 13
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 10:15:12] mlp_actions: End of event 13
[1:ml1:Folyoso] [NTC] [EVT] [Feb 02 10:57:06] event_newfile: File of type 8 saved to: /mnt/camera-storage/FOLYOSO_1-14-2023-02-02_10-57-05-Treshold-5000-ChangedPixels-10859-NoiseLevel-11.mkv
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 10:57:06] motion_detected: Motion detected - starting event 14
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 10:58:06] mlp_actions: End of event 14
[1:ml1:Folyoso] [NTC] [EVT] [Feb 02 11:08:34] event_newfile: File of type 8 saved to: /mnt/camera-storage/FOLYOSO_1-15-2023-02-02_11-08-33-Treshold-5000-ChangedPixels-59585-NoiseLevel-11.mkv
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 11:08:34] motion_detected: Motion detected - starting event 15
[2:ml2:Iroda] [NTC] [ALL] [Feb 02 11:09:32] mlp_actions: End of event 11
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 11:09:32] mlp_actions: End of event 15
[2:ml2:Iroda] [NTC] [ALL] [Feb 02 11:09:32] motion_loop: Thread exiting
[1:ml1:Folyoso] [NTC] [ALL] [Feb 02 11:09:32] motion_loop: Thread exiting
[2:ml2:Iroda] [NTC] [NET] [Feb 02 11:09:33] netcam_rtsp_cleanup: Normal resolution: Shut down complete.
[1:ml1:Folyoso] [NTC] [NET] [Feb 02 11:09:33] netcam_rtsp_cleanup: Normal resolution: Shut down complete.
[0:motion] [NTC] [ALL] [Feb 02 11:09:34] main: Threads finished
[0:motion] [NTC] [ALL] [Feb 02 11:09:34] main: Motion terminating
[0:motion] [NTC] [ALL] [Feb 02 11:09:34] motion_remove_pid: Closing logfile (/var/log/motion/motion.log).
```

**Comment by Mr-Dave** [comment]
This just sounds like PI performance.

**Comment by tosiara** [comment]
I agree, that looks like a performance issue, check CPU usage. Try to run only one camera with the lowest resolution possible (ex, use sub-stream) and you will notice that the issue will go away.

ARM devices are not capable of decoding 720p RTSP H264 and encoding at the same time

**Reply by anonimusz20220712** [reply]
OH, OK, I understand, I will check the cpu performance!

Thank you!

**Reply by anonimusz20220712** [reply]
This is by Munin, but maybe the frequency(time intervals) is not high enough:

![image](https://user-images.githubusercontent.com/124486842/217228890-d95f1427-9c3e-4040-8ba5-f6fde9ada90b.png)

It was never too high :(

**Reply by anonimusz20220712** [reply]
The load was neither

![image](https://user-images.githubusercontent.com/124486842/217228986-2ab9f291-39b7-45b1-b130-d17842d5a370.png)

**Reply by tosiara** [reply]
you want to say that those graphs are showing Rpi4 running and recording 2x 720p RTSP cameras? please show the graph since the beginning of february

**Reply by anonimusz20220712** [reply]
This is that day:

![image](https://user-images.githubusercontent.com/124486842/217248312-1a8e092c-1881-4122-81a4-9291535a15f4.png)

![image](https://user-images.githubusercontent.com/124486842/217248182-50e173d6-bd20-4acc-a044-e38a9eb52afc.png)

**Reply by anonimusz20220712** [reply]
Tha load wasn't too high:

![image](https://user-images.githubusercontent.com/124486842/217248583-d80d096c-13e9-403d-b6a9-abd2a986dd7e.png)

**Comment by tosiara** [comment]
Can you reproduce the issue with one camera and upload a full log with `-d 9` ?

**Reply by anonimusz20220712** [reply]
OK, I will check it out!
